File: Data/Models/TransferNoteDetail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Nebula.Data.Models
{
    public class TransferNoteDetail
    {
        public int Id { get; set; }
        public int? TransferNoteId { get; set; }

        [JsonIgnore]
        [ForeignKey("TransferNoteId")]
        public TransferNote TransferNote { get; set; }

        public int? ProductId { get; set; }
        [MaxLength(250)] public string Description { get; set; }
        public decimal? Price { get; set; }
        public decimal? Quantity { get; set; }
        public decimal? Amount { get; set; }
    }
}

File: Angular/src/app/inventory/interfaces/transfer-note-detail.ts
export class TransferNoteDetail {
  constructor(
    public id: string = '',
    public transferNoteId: number = 0, // Id Nota de Transferencia.
    public productId: number = 0, // Id del producto.
    public description: string = '', // Descripción del producto.
    public price: number = 0, // Precio del producto.
    public quantity: number = 0, // Cantidad del producto.
    public amount: number = 0) {
  }
}
